test/test_dht.cpp

```diff
@@ -2059,5 +2121,103 @@ TORRENT_TEST(routing_table_extended)
 #endif
 }
 
+TORRENT_TEST(read_only_node)
```

Comment:
it seems like the code in here is duplicated twice. Do you think it would be possible to factor out the test and call it twice (once with read-only on and one with it off)?